### PR TITLE
Stats table redux

### PR DESCRIPTION
**Pull request by maddiedierker**
Refactors the new `StatsTableWithData` in the new dashboard-served teacher dashboard to store/receive all of its data from redux.

### CODE REVIEW

File: apps/src/templates/teacherDashboard/TeacherDashboard.jsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ export default class TeacherDashboard extends Component {
         <Switch>
           <Route
             path="/stats"
-            component={props => <StatsTableWithData {...props} sectionId={sectionId}/>}
+            component={props => <StatsTableWithData {...props}/>}
```
Comment (davidsbailey) [inline apps/src/templates/teacherDashboard/TeacherDashboard.jsx]: can we omit `{...props}` now that all the props of `StatsTableWithData` are provided by redux?

File: apps/src/templates/teacherDashboard/StatsTable.jsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ class StatsTable extends Component {
 
   studentsWithCompletedLevelCount = () => {
     const {section, studentsCompletedLevelCount} = this.props;
-    return section.students.map(student => ({
+    return (section.students || []).map(student => ({
```
Comment (davidsbailey) [inline apps/src/templates/teacherDashboard/StatsTable.jsx:30]: should `section` be a required proptype? or should we check presence of `section` here before looking at `section.students`? seems like we should do one or the other to clarify whether `section` is required or not.

**Review by davidsbailey — APPROVED** [review]
Nice clean-up Maddie! It's great to see this state all move into redux.

One thing not addressed in this PR is UI tests. We do have one place that the stats table is loaded in UI tests here, but we don't verify that anything inside this react component has actually loaded: 
https://github.com/code-dot-org/code-dot-org/blob/fa7f5a64124bded4dfb689375b1404caf123788e/dashboard/test/ui/features/teacher_dashboard.feature#L21-L25

I think this would be a good time to add a line to the UI test which verifies that Sally's name eventually appears in the table, or something similar, to verify that we're displaying the data we get from the server. OK to do in a separate PR if you need, though.

**Comment by maddiedierker** [comment]
@davidsbailey agreed about UI tests. the current UI tests are testing the teacher dashboard components served from pegasus, so we will have to do some refactoring to those tests once we swap over to the dashboard-served teacher dashboard components

**Comment by davidsbailey** [comment]
oh right! yes, no need to add UI tests for this now, then.

**Pull request merged**